### DISCUSSION TITLE
Problem running directus on cPanel nodejs (sharp)

### DISCUSSION THREAD

**Discussion by DanielProp** [directus/directus#15917] [Archive]
Hi everyone,

I'm trying to deploy my directus app to a shared hosting I have, with cPanel that allows for node.js apps.
I setup the app using the cPanel node.js UI using "node_modules\directus\dist\start.js" as the Application startup file (#3406)
Accessing through SSH, I enter the "virtual environment" where I have a little more control. when I run `npx directus start` I get the following error:

```
sharp: Installation error: Use with glibc 2.12 requires manual installation of libvips >= 8.12.2
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! [sharp@0.30.7](mailto:sharp@0.30.7) install: '(node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)'
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the [sharp@0.30.7](mailto:sharp@0.30.7) install script.
```

So the problem is that I can't install sharp because I don't have root access. I've tried `npm install --unsafe-perm` but it gives me the same output.

Is there any way I can get around this? Is anyone having any luck hosting directus on shared hosting with cPanel? Or am I just gonna have to give up and upgrade to a vps or a different hosting?

Thanks in advance for taking the time.

**Comment by lukas-schaetzle** [comment]
There are quite some people, me included, successfully running Directus on a shared hosting plan with CPanel or Plesk (which is basically the same as far as I can tell).
However, looking at your error I think your hosting provider simply uses a very old glibc version (needed for building sharp and also argon2 btw). So you either have to ask your provider to update the package or do it yourself. (But as you said you would need root for that which is not the case on shared hosting plans.)

**Reply by DanielProp** [reply]
Thank you for your answer @Tummerhore.
I did reach out to the provider, but they just suggested I upgrade to a VPS. And I do not have permissions to install it myself.
I guess I'm gonna try a different provider. I'm looking at netcup, which the directus docs mention [here](https://docs.directus.io/self-hosted/installation/plesk.html)
I'd appreciate more options. If you don't mind sharing what's working for you. Thanks!

**Reply by lukas-schaetzle** [reply]
Yeah, that's often problematic with shared hosting plans. Most of the time you can't say if it will work there until you really try it out.

I can only talk about netcup since that is what I'm using (that mention in the docs is from me :) ). I also considered a2 hosting but in the end didn't try it out. If you wanna go with netcup: Make sure to use at least the plan "Webhosting 4000" for node.js support. I'm using that plan and the performance is good enough for a small website running with directus (+nextcloud) in the background. If you try another hoster and it works there, please feel free to add it to the docs :)

Please note that updating directus on shared hosts sometimes does not work without problems (e.g. there were problems with natively built dependencies or a directus update did not work with Plesk/CPanel for a while in the past). But at least the plans are cheap and so far the Directus community was able to solve all the problems sooner or later.

**Reply by DanielProp** [reply]
I'll play it safe and try netcup. Thanks a lot for all the info!
Just for the record, the hosting that did NOT work for me was shared hosting by namecheap.

**Reply by jojojojojoj5564656465465** [reply]
have you found a solution to install it ?

**Reply by DanielProp** [reply]
Unfortunately not. I ended up switching hosting provider to netcup.

**Reply by rin4ik** [reply]
uploading package.lock file directly to nodevenv/lib helped

**Reply by wasifAL** [reply]
I dont know if you've resolved it. Just sharing my experience.

Building directus within the same os version as used in the shared hosting shall resolve this issue. You can try it via a docker image / VM if you're developing using a different OS tree.
